insights/parsers/numeric_user_group_name.py

```diff
@@ -3,26 +3,38 @@
 =========================================================================================
 
 This parser reads the output of ``/usr/bin/grep -c '^[[:digit:]]' /etc/passwd /etc/group``
-and detects whether it is 0 or nonzero.
+and detects whether there are any user or group names that begin with a digit.
 
-The grep command returns the number of matches. It matches for user and group names that
-begin with a number. If it returns 0, then there are no such names, otherwise it returns
-the number of matches (names). The parser only detects whether it is 0 or nonzero so as
-to minimize the amount of collected data.
+The grep command returns the number of matches for each file. It matches for user and
+group names that begin with a number. The parser only detects whether the sum is 0 or
+nonzero so as to minimize the amount of collected data.
 
-The boolean is available in the ``data`` attribute but it should not be accessed directly
-(encapsulation). The parser provides methods for comparisons with bool and casting to bool.
+The boolean answers are available through the getters ``has_numeric_user_group()``,
+``has_numeric_user()``, ``has_numeric_group()`` and ``bad_data()``.
 
-Sample input::
+Names starting with a digit can be handled incorrectly by some software. Purely numeric
+user names can be handled incorrectly by some software and requires special care when
+used with utilities that accept both user/group IDs and names (like chown, chmod). In
+case such names exist on the system, it is advisable to review whether the used software
+(namely 3rd party software and shell scripts) handles the names correctly.
 
-    3
+https://access.redhat.com/solutions/3103631
+
+Sample input:
+
+/etc/passwd:3
+/etc/group:0
 
 Examples:
 
-    >>> bool(shared[NumericUserGroupName])
+    >>> shared[NumericUserGroupName].bad_data()
+    False
+    >>> shared[NumericUserGroupName].has_numeric_user_or_group()
```

Comment:
The reason was to keep it encapsulated 1) because a getter cannot change the attribute, 2) in case we ever want to change the internal implementation. But you're right, this is so simple any refactoring in future will be simple as well.